commands/setUser.js

```diff
@@ -2,6 +2,106 @@ const sqlite = require('sqlite');
 const sqlite3 = require('sqlite3');
 const config = require('../config.' + process.env.NODE_ENV_MODE);
 
+const audit = async function(userid, server, alumni_role, officer_role, alumni_officer_role) {
+  // open db
+  let db = await sqlite.open({
+    filename: config.db_path,
+    driver: sqlite3.Database,
+  });
+
+  // check if user is verified
+  let row = null;
+  try {
+    row = await db.get(
+      `
+     SELECT
+         userid, nickname
+     FROM users
+     WHERE
+         userid = ?`,
+      [userid]
+    );
+  } catch (e) {
+    console.error(e.toString());
+    await db.close();
+    return [{ message: e.toString() }, null];
+  }
+
+  if (!row) {
+    await db.close();
+    return [
+      null,
+      `Sorry, I don't think you're verified!.
+Use \`/iam\` to verify your email address.`,
+    ];
+  }
+
+  let today = new Date();
+  let month = String(today.getMonth() + 1).padStart(2, '0');  // 0 indexing, i.e. January is 0
+  let year = today.getFullYear();
+  //  if the command is run July 2021, it would catch everyone who had just graduated
+  // while if the command was run in May, it would not affect the students who have yet to graduate
+  if (month < 6)
+    year--;
+
+  let rows = null;
+  try {
+    rows = await db.all(
+      `
+    SELECT
+        userid
+    FROM users
+    WHERE
+        grad_year <= ?
+    AND
+        affiliation = 'student'`,
+      [year]
+    );
+  } catch (e) {
+    console.error(e.toString());
+    await db.close();
+    return [{ message: e.toString() }, null];
+  }
+
+  try {
+    await db.run(
+      `
+      UPDATE
+          users
+      SET
+          affiliation = 'alumni'
+      WHERE
+          grad_year <= ?
+      AND
+          affiliation  = 'student'`,
+      [year]
+    );
+  } catch (e) {
+    console.error(e.toString());
+    await db.close();
+    return [{ message: e.toString() }, null];
+  }
+
+  await db.close();
+
+  rows.forEach(async(user) => {
+    let id = user['userid'];
+    let server_member = await server.members.fetch(id);
```

Comment:
Should we handle the case if the member isn't found? In case the database is out of sync with the server.